From 243d2e366e8fc11719f9b5283a0a68f39a992c65 Mon Sep 17 00:00:00 2001
From: tista500 <tista.gma500@gmail.com>
Date: Tue, 10 Oct 2017 04:12:53 +0900
Subject: [PATCH] Gtk-3.22&4.0: [GtkWindow] Exclude GtkDialogs when putting
 specific tiled-edges

I'm really confused with the report: #530.
Because GtkDialog should never be turned to tiled-mode (via .tiled-* classes),
But Reporter's shots seem to draw our tiled-edge specific drop-shadow
parameters in GtkDialogs. Mystery...

So I'm not quite sure this could make sense since this was not needed
semantically.
---
 gtk/sass/3.22/_common.scss | 6 ++++--
 gtk/sass/4.0/_common.scss  | 6 ++++--
 2 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/gtk/sass/3.22/_common.scss b/gtk/sass/3.22/_common.scss
index 009259658..9f82e8e8e 100644
--- a/gtk/sass/3.22/_common.scss
+++ b/gtk/sass/3.22/_common.scss
@@ -5940,7 +5940,9 @@ decoration {
                     ('-left', $z-depth-2-backdrop-tiled-left),
                     ('-right' $z-depth-2-backdrop-tiled-right) {
     // draw empty shadows + window edges when tiled
-    .tiled#{$_constraint} & {
+    // FIXME: window node was not needed semantically, but Solus build seems
+    // to draw tiled-edges in GtkDialogs. Mystery...
+    window.tiled#{$_constraint} & {
       border-radius: 0;
       box-shadow: #{$_shadow};
     }
@@ -5963,7 +5965,7 @@ decoration {
                     ('-left', $z-depth-1-backdrop-tiled-left),
                     ('-right' $z-depth-1-backdrop-tiled-right) {
     // draw empty shadows + window edges when tiled
-    .ssd.tiled#{$_constraint} & { box-shadow: #{$_shadow}; }
+    window.ssd.tiled#{$_constraint} & { box-shadow: #{$_shadow}; }
   }
 
   .csd.popup & {
diff --git a/gtk/sass/4.0/_common.scss b/gtk/sass/4.0/_common.scss
index 02b1c3862..15cd8c01e 100644
--- a/gtk/sass/4.0/_common.scss
+++ b/gtk/sass/4.0/_common.scss
@@ -5834,7 +5834,9 @@ decoration {
                     ('-left', $z-depth-2-backdrop-tiled-left),
                     ('-right' $z-depth-2-backdrop-tiled-right) {
     // draw empty shadows + window edges when tiled
-    .tiled#{$_constraint} & {
+    // FIXME: window node was not needed semantically, but Solus build seems
+    // to draw tiled-edges in GtkDialogs. Mystery...
+    window.tiled#{$_constraint} & {
       border-radius: 0;
       box-shadow: #{$_shadow};
     }
@@ -5857,7 +5859,7 @@ decoration {
                     ('-left', $z-depth-1-backdrop-tiled-left),
                     ('-right' $z-depth-1-backdrop-tiled-right) {
     // draw empty shadows + window edges when tiled
-    .ssd.tiled#{$_constraint} & { box-shadow: #{$_shadow}; }
+    window.ssd.tiled#{$_constraint} & { box-shadow: #{$_shadow}; }
   }
 
   .csd.popup & {
